Lib/test/test_msvcrt.py

```diff
@@ -62,33 +61,45 @@ def test_get_osfhandle(self):
 
 class TestConsoleIO(unittest.TestCase):
     def test_kbhit(self):
-        h = msvcrt.get_osfhandle(sys.stdin.fileno())
-        flush_console_input_buffer(h)
-        self.assertEqual(msvcrt.kbhit(), 0)
+        # Run test in a seprated process to avoid stdin conflicts.
+        # See: gh-110147
+        cmd = [sys.executable, '-c',
+               'import msvcrt;'
+               'assert msvcrt.kbhit() == 0;'
+               ]
+        subprocess.run(cmd, check=True, capture_output=True)
```

Comment:
Does creating a new process separate the console between the parent and child process?

Do you think that passing CREATE_NEW_CONSOLE could make the test more reliable?

https://learn.microsoft.com/en-us/windows/win32/procthread/process-creation-flags

> The new process has a new console, **instead of inheriting its parent's console (the default)**. For more information, see [Creation of a Console](https://learn.microsoft.com/en-us/windows/console/creation-of-a-console).
>  This flag cannot be used with DETACHED_PROCESS.

I suggest you creating a helper function for ``subprocess.run(cmd, check=True, capture_output=True)`` and add CREATE_NEW_CONSOLE flag.

Something like:

```py
startupinfo = subprocess.STARTUPINFO(dwflags=subprocess.CREATE_NEW_CONSOLE)
subprocess.run(..., startupinfo=startupinfo)
```

Reply:
Updated, the `CREATE_NEW_CONSOLE` should be passwd as `creationflags`.